Applications/Spire/Source/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,106 @@
+#include "Spire/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.hpp"
+
+using namespace boost;
+using namespace boost::icl;
+using namespace boost::posix_time;
+using namespace Nexus;
+using namespace Spire;
+
+CachedOrderImbalanceIndicatorModel::CachedOrderImbalanceIndicatorModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source)
+    : m_source_model(std::move(source)) {
+  auto [connection, promise] = m_source_model->subscribe(
+    [=] (const auto& imbalance) {
+      on_imbalance_published(imbalance);
+    });
+  m_subscription_connection = std::move(connection);
+  m_subscribe_promise = std::move(promise);
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    CachedOrderImbalanceIndicatorModel::load(
+    const TimeInterval& interval) {
+  if(boost::icl::contains(m_intervals, interval)) {
+    return load_from_cache(interval);
+  }
+  return load_from_model(interval);
+}
+
+SubscriptionResult<optional<Nexus::OrderImbalance>>
+    CachedOrderImbalanceIndicatorModel::subscribe(
+    const OrderImbalanceSignal::slot_type& slot) {
+  return m_source_model->subscribe(slot);
+}
+
+std::shared_ptr<OrderImbalanceChartModel>
+    CachedOrderImbalanceIndicatorModel::get_chart_model(
+    const Security& security) {
+  throw std::runtime_error("method not implemented");
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    CachedOrderImbalanceIndicatorModel::load_from_cache(
+    const TimeInterval& interval) {
+  return m_cache.load(interval);
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    CachedOrderImbalanceIndicatorModel::load_from_model(
+    const TimeInterval& interval) {
+  auto promises = std::vector<QtPromise<std::vector<OrderImbalance>>>();
+  auto load_intervals = interval_set<ptime>(interval);
+  load_intervals = load_intervals - m_intervals;
```

Comment:
Rename ```load_intervals``` to ```loaded_intervals```, "load" is a verb and only functions are named after verbs, variables are named after nouns.

Reply:
I renamed it to 'unloaded_intervals' to make it clearer; I was trying to name the unloaded portions.